### PR TITLE
Add shell script to scan for IOCs via SSM

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR adds an "extras" directory containing a shell script that can be used to scan a list of AWS EC2 instances via [SSM](https://docs.aws.amazon.com/systems-manager/latest/userguide/what-is-systems-manager.html) for indicator of compromise strings.

## 💭 Motivation and context ##

This script is useful for cases where the instances you want to scan are only accessible via SSM.

## 🧪 Testing ##

I ran the script with a variety of inputs and verified that it worked as expected.  I also tested the script with no arguments, with invalid arguments (e.g. a non-existent IOC file), and with `-h` and with `--help` and confirmed that the usage text was displayed as intended in those cases.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

### CODE REVIEW

File: extras/ioc_scan_by_host.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# tee -i: Ignore SIGINT signals
+exec > >(tee -ai "$logfile")
+exec 2> >(tee -ai "$logfile" >&2)
+
+# Get list of arguments passed to script, but ignore the first two (script name
+# and IOC file); the rest are the instance IDs.
+instances=("${@:2}")
+
+echo "IOC List is: \"${iocList[*]}\""
+echo "Instances are: \"${instances[*]}\""
```
Comment (jsf9k) [inline extras/ioc_scan_by_host.sh]: `echo` doesn't require that the text to be printed be in quotes, so I think there are several places in this script where you can simplify your quoting by removing the outermost double quotes, e.g.:
```suggestion
echo IOC List is: "${iocList[*]}"
echo Instances are: "${instances[*]}"
```

**Review by jsf9k — COMMENTED** [review]
I have one comment that applies all over the show in `ioc_scan_by_host.sh`.  Let me know if you agree.

File: extras/ioc_scan_by_host.sh (hunk 1 above)
Comment (dav3r) [reply extras/ioc_scan_by_host.sh]: Yup, thanks for that reminder!  See https://github.com/cisagov/ioc-scanner/pull/45/commits/b1c498908f054a69840e5ac94b40a5aeec4967fb.

File: extras/ioc_scan_by_host.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# and IOC file); the rest are the instance IDs.
+instances=("${@:2}")
+
+echo IOC List is: "${iocList[*]}"
+echo Instances are: "${instances[*]}"
+
+# Loop through all instance IDs
+for instance_id in "${instances[@]}"; do
+  # "echo -e" allows us to use \n for newlines.
+  echo -e "\nSearching $instance_id:"
```
Comment (jsf9k) [inline extras/ioc_scan_by_host.sh]: This is the usual way of doing it.
```suggestion
  echo
  echo Searching "$instance_id":
```

**Review by jsf9k — APPROVED** [review]
I have one minor suggestion, but my approval stands even if you choose to disregard it.

File: extras/ioc_scan_by_host.sh (hunk 2 above)
Comment (dav3r) [reply extras/ioc_scan_by_host.sh]: Works for me!

File: README.md [hunk 3]
```diff
@@ -82,6 +82,30 @@ ansible --inventory=hosts-file cool-servers \
...
 
+To scan for indicator strings on AWS instances that are accessible via
+[SSM](https://docs.aws.amazon.com/systems-manager/latest/userguide/what-is-systems-manager.html),
+the `ioc_scan_by_host.sh` shell script has been provided in the `extras`
+directory:
+
+```console
+$ ./extras/ioc_scan_by_host.sh my-ioc-strings.txt i-0123456789abcdef0
+
+IOC List is: "192.168.1[.]1 sketchy-site[.]org IOC_HASH_1 IOC_HASH_2"
```
Comment (mcdonnnj) [inline README.md]: Just as a discussion point does mentioning hashes make sense here? The Python logic in this project generates hashes of files and checks those while this script searches for strings in log files. I don't want to conflate the functionality of the two components as the script added in this PR does no hash generation. Maybe something more like:

```suggestion
IOC List is: "192.168.1[.]1 sketchy-site[.]org IOC_STRING_1 IOC_STRING_2"
```

File: README.md [hunk 4]
```diff
@@ -82,6 +82,30 @@ ansible --inventory=hosts-file cool-servers \
...
+Instances are: "i-0123456789abcdef0"
+
+Searching i-0123456789abcdef0:
+
+Starting session with SessionId: iam.username-0123456789abcdef0
+0 found for 192.168.1[.]1
+0 found for sketchy-site[.]org
+0 found for IOC_HASH_1
+0 found for IOC_HASH_2
```
Comment (mcdonnnj) [inline README.md]: If the above suggestion is accepted this should be as well.

```suggestion
0 found for IOC_STRING_1
0 found for IOC_STRING_2
```

File: README.md (hunk 3 above)
Comment (dav3r) [reply README.md]: That is an excellent point.  I will make that change shortly.

**Review by mcdonnnj — APPROVED** [review]
:shipit:

**Pull request merged**